programs/locker/src/instructions/v2/claim.rs

```diff
@@ -0,0 +1,75 @@
+use anchor_spl::memo::Memo;
+use anchor_spl::token_interface::{
+    Mint, TokenAccount, TokenInterface,
+};
+
+use crate::*;
+use crate::constants::transfer_memo;
+use crate::util::{MemoTransferContext, transfer_to_user_v2};
+
+#[event_cpi]
+#[derive(Accounts)]
+pub struct ClaimV2<'info> {
+    #[account(
+        mut, 
+        has_one = recipient,
```

Comment:
should add has_one = token_mint?

Reply:
I think we also need `has_one = recipient` to ensure the one who is claiming is the valid recipient, no ?

Reply:
updated